assets/tests/QuestionSchemaValidationService.js

```diff
@@ -60,6 +60,23 @@ tie.factory('QuestionSchemaValidationService', [
           angular.isArray(question.getTasks()) &&
           question.getTasks().length > 0);
       },
+      verifyAtLeastOneBuggyOutputTestExists: function(question) {
+        return question.getTasks().some(function(task) {
+          return task.getBuggyOutputTests().length > 0;
+        });
+      },
+      verifyAllBuggyOutputTestMessagesAreUnique: function(question) {
+        var messages = new Set();
+        return question.getTasks().every(function(task) {
+          return task.getBuggyOutputTests().every(function(test) {
```

Comment:
Shouldn't the `messages` set just be within the context of a single task? If so, it might also be simpler to just dump all the messages into a set and verify that the set's size equals the length of the list of messages.